## Подготовительные работы.

* Установка необходимых для анализа библиотек.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
display('Все библиотеки импортированы')

'Все библиотеки импортированы'

* Загрузка датасетов.

In [2]:
categories = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-categories.csv')
orders = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-orders.csv')
order_items = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-order_items.csv')
products = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-products.csv')
reviews = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-reviews.csv')
transactions = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-transactions.csv')
users = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-users.csv')
display('Все датасеты загружены')

'Все датасеты загружены'

## Знакомство с датасетами.

### Знакомство с таблицей `categories`.

In [3]:
# Выводим информацию о датафрейме, первые 5 строк и количество полных дубликатов
display(categories.info())
display(categories.head(5))
display(f'Полных дубликатов: {categories.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1800 entries, 0 to 1799
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   category_id         1800 non-null   int64  
 1   name                1798 non-null   object 
 2   parent_category_id  915 non-null    float64
dtypes: float64(1), int64(1), object(1)
memory usage: 42.3+ KB


None

,category_id,name,parent_category_id
0,0,Available,NaN
1,1,Imagine,NaN
2,2,Civil,1238.0
3,3,Six,243.0
4,4,NaN,NaN


'Полных дубликатов: 0'

* Таблица `categories` состоит из 3 колонок и 1800 строк;
* Присутствуют пропущенные значения в поле `parent_category_id` - объясняется тем, что соответствующее значение в поле `category_id` уже является родительской;
* ⚠️ Присутствуют пропущенные значения в поле `name` - можно добавить заглушку;
* Отсутствуют полные дубликаты;

### Знакомство с таблицей `orders`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(orders.info())
display(orders.head(5))
display(f'Полных дубликатов: {orders.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 505400 entries, 0 to 505399
Data columns (total 5 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   order_id      505400 non-null  object 
 1   buyer_id      505400 non-null  object 
 2   order_date    505400 non-null  object 
 3   status        505400 non-null  object 
 4   total_amount  505400 non-null  float64
dtypes: float64(1), object(4)
memory usage: 19.3+ MB


None

,order_id,buyer_id,order_date,status,total_amount
0,2299f3d1-f01e-4d6a-ad54-c9a74bff0c02,bc69766f-aab2-469c-b24f-4b00151aa519,2025-04-27 10:27:18.000,paid,4674.51
1,186fdd71-4ddf-4198-9162-c71febc7bc05,e421a631-2faf-4810-8a81-7f377d728eff,2025-02-24 15:08:42.000,new,1575.70
2,4ab13c5e-924a-43f1-a77a-17d28bd0f6f6,319b30cf-0ef0-4e8e-a1bc-ca83136b59cc,2024-01-16 04:13:06.000,canceled,2048.65
3,11bd972e-20a2-4ce7-96fb-bb4ed47c4614,fbff34d4-644d-456b-9331-833d1aa58a9e,2025-04-09 09:32:46.000,shipped,5267.75
4,2ff312f2-e2a2-4b17-a94c-e29211206728,7651578e-fa3b-4888-9000-2d1e607712cf,2025-06-08 12:58:51.000,shipped,1560.81


'Полных дубликатов: 0'

* Таблица `orders` состоит из 5 колонок и 505400 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `order_date` необходимо преобразовать к datatype типу данных.

In [ ]:
# Преобразование типа данных
orders['order_date'] = pd.to_datetime(orders['order_date'])
display(orders.dtypes)

order_id                object
buyer_id                object
order_date      datetime64[ns]
status                  object
total_amount           float64
dtype: object

* ✅ Поле `order_date` преобразовано к datatype типу данных.

### Знакомство с таблицей `order_items`.

In [12]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(order_items.info())
display(order_items.head(5))
display(f'Полных дубликатов: {order_items.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1516751 entries, 0 to 1516750
Data columns (total 5 columns):
 #   Column               Non-Null Count    Dtype  
---  ------               --------------    -----  
 0   order_item_id        1516751 non-null  object 
 1   order_id             1516751 non-null  object 
 2   product_id           1516751 non-null  object 
 3   quantity             1516751 non-null  int64  
 4   price_at_order_time  1516751 non-null  float64
dtypes: float64(1), int64(1), object(3)
memory usage: 57.9+ MB


None

,order_item_id,order_id,product_id,quantity,price_at_order_time
0,4633224a-eb5b-4c2e-a568-8e5f3676d839,98895f4d-611c-41d2-88f5-78a5b58c78b5,ba66791e-60f0-43f2-8d1d-a84e6b047657,2,279.71
1,d68dbbb4-0323-447f-9afb-9e2def518732,fbc1d88b-aa81-43e6-b156-b2f4a8d9ec8a,e0a10065-76ac-4557-8638-2dfe422f415f,1,314.66
2,d38e7218-e244-4afe-97f9-0ba0492eaf57,fbc1d88b-aa81-43e6-b156-b2f4a8d9ec8a,0d122bb3-888c-4fcc-8609-057c9c3ed2e8,3,145.99
3,ab970c22-4625-4944-9e73-5fdbc04a9928,fbc1d88b-aa81-43e6-b156-b2f4a8d9ec8a,ef894d8e-a749-43a8-9486-6bbfe247364a,1,955.55
4,0ffc4785-c391-48c6-932d-e85e0c552c97,928d50b8-ab93-46dc-ac42-3e57224bd0b6,ac47bd16-2356-4086-b94d-3f1b5d090174,1,585.98


'Полных дубликатов: 0'

* Таблица `order_items` состоит из 5 колонок и 1516751 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты.

### Знакомство с таблицей `reviews`.

In [13]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(reviews.info())
display(reviews.head(5))
display(f'Полных дубликатов: {reviews.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 310100 entries, 0 to 310099
Data columns (total 6 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   review_id    310100 non-null  object
 1   user_id      310100 non-null  object
 2   product_id   310100 non-null  object
 3   rating       310100 non-null  int64 
 4   comment      310100 non-null  object
 5   review_date  310100 non-null  object
dtypes: int64(1), object(5)
memory usage: 14.2+ MB


None

,review_id,user_id,product_id,rating,comment,review_date
0,92e9eb35-0063-448c-a407-47e99530eb99,5fcf22b3-2788-4102-a3f4-7721d92d923f,77ac4b77-b310-47cc-873b-66e6a0ced5a1,2,Another gun wide read former.,2025-02-12 05:12:04.000
1,78864309-374c-4499-adbd-63e7ed1f604e,0f1a1211-70c9-42ec-8dca-1c56b9bd8206,13c4fcfe-bd79-49ac-a666-1e19507b9896,3,Ahead offer such when.,2024-07-05 16:19:28.000
2,239fea6e-b3aa-4786-8bff-563bae3400c4,f59d2104-5752-4b16-91be-b2dec691c978,5dcf670c-c917-43e4-859b-86cbc78cfef1,2,Type require now appear.,2024-01-29 02:48:17.000
3,b7e67361-947c-48cd-8586-be4806a0d460,5bce0123-4c46-4aef-a71a-02b276a47f57,941568fb-88e9-40ad-8a89-37457c667fd3,2,Attorney beyond newspaper year your.,2025-02-03 13:28:40.000
4,b6a009ff-f752-497a-8b3f-929abc73007c,04a6dd69-1ea0-4a41-8c6a-93a8c8b71b8f,79ba5821-ac61-4902-9cc8-01bdfe29a8e9,4,Your hard law loss relationship however husband.,2024-07-22 05:21:30.000


'Полных дубликатов: 0'

* Таблица `reviews` состоит из 6 колонок и 310000 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `review_date` необходимо преобразовать к datatype типу данных.

In [14]:
# Преобразование типа данных
reviews['review_date'] = pd.to_datetime(reviews['review_date'])
display(reviews.dtypes)

review_id              object
user_id                object
product_id             object
rating                  int64
comment                object
review_date    datetime64[ns]
dtype: object

* ✅ Поле `review_date` преобразовано к datatype типу данных.

### Знакомство с таблицей `users`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(users.info())
display(users.head(5))
display(f'Полных дубликатов: {users.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 111300 entries, 0 to 111299
Data columns (total 7 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   user_id            111300 non-null  object        
 1   name               111300 non-null  object        
 2   email              111300 non-null  object        
 3   phone              111300 non-null  object        
 4   user_type          111300 non-null  category      
 5   registration_date  111300 non-null  datetime64[ns]
 6   is_active          111300 non-null  bool          
dtypes: bool(1), category(1), datetime64[ns](1), object(4)
memory usage: 4.5+ MB


None

,user_id,name,email,phone,user_type,registration_date,is_active
0,d447326f-e55e-4eb1-8401-222deb8539fd,Patrick Sanchez,jillrhodes@example.net,296-500-1338,buyer,2024-10-19 06:39:46,True
1,bf50f4dd-8e31-47e9-b3f4-f0ffcfa9a167,Kimberly Robinson,lisa02@example.net,(254)923-5116x15594,seller,2024-12-17 18:27:22,True
2,3290a5be-9ced-4e7b-9068-b3a884b4f021,Thomas Miller,maria95@example.net,610.934.1316,buyer,2024-07-31 02:53:58,True
3,630a0e61-8be5-4a18-a7e6-7e22c56a9a21,Carla Kelly,jacqueline19@example.net,(783)227-6483,buyer,2024-10-16 08:54:00,True
4,e9e71002-986f-496c-a4c0-12632edf8ea0,Renee Morales,clarksherri@example.net,837-767-2423x88496,seller,2023-12-05 12:38:55,True


'Полных дубликатов: 0'

* Таблица `users` состоит из 7 колонок и 111300 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `user_type` необходимо преобразовать к category типу данных;
* ⚠️ Поле `registration_date` необходимо преобразовать к datatype типу данных.

In [ ]:
# Преобразование типа данных
users['user_type'] = users['user_type'].astype('category')
users['registration_date'] = pd.to_datetime(users['registration_date'])
display(users.dtypes)

user_id                      object
name                         object
email                        object
phone                        object
user_type                  category
registration_date    datetime64[ns]
is_active                      bool
dtype: object

* ✅ Поле `user_type` преобразовано к category типу данных;
* ✅ Поле `registration_date` преобразовано к datatype типу данных.

### Знакомство с таблицей `transactions`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(transactions.info())
display(transactions.head(5))
display(f'Полных дубликатов: {transactions.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 811200 entries, 0 to 811199
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   transaction_id    811200 non-null  object 
 1   user_id           811200 non-null  object 
 2   amount            811200 non-null  float64
 3   transaction_type  811200 non-null  object 
 4   transaction_date  811200 non-null  object 
dtypes: float64(1), object(4)
memory usage: 30.9+ MB


None

,transaction_id,user_id,amount,transaction_type,transaction_date
0,2bb73129-84ce-48f0-9550-e123ef9aec69,879926c8-c9cb-4ee5-a4b6-108dcbae85be,121.94,payout,2025-03-13 20:49:20.000
1,7ea0e3cf-ff3a-4476-a889-1a7f7e31bb8d,8a5f444c-c7bd-46ff-bd15-dd765bd77115,900.36,payment,2025-04-29 18:01:23.000
2,09db96bb-8ecf-4ad4-a24d-5aa85b6e2c20,ea224611-be68-4151-b681-5e19a019e9c3,-283.42,payout,2024-12-24 12:13:00.000
3,ac97bcac-59a1-41ef-b157-076426af2e19,f3fb3feb-8623-438e-9ed3-5e98060f7d91,210.62,payout,2025-04-05 07:15:09.000
4,2ce23859-0181-4c0f-b3cf-c25acca56467,17da5324-6730-4808-a341-43a9bf74664c,744.45,payout,2024-09-19 20:18:40.000


'Полных дубликатов: 0'

* Таблица `transactions` состоит из 5 колонок и 811200 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `transaction_type` необходимо преобразовать к category типу данных;
* ⚠️ Поле `transaction_date ` необходимо преобразовать к datatype типу данных.

In [ ]:
# Преобразование типа данных
transactions['transaction_type'] = transactions['transaction_type'].astype('category')
transactions['transaction_date'] = pd.to_datetime(transactions['transaction_date'])
display(transactions.dtypes)

transaction_id              object
user_id                     object
amount                     float64
transaction_type          category
transaction_date    datetime64[ns]
dtype: object

* ✅ Поле `transaction_type` преобразовано к category типу данных;
* ✅ Поле `transaction_date` преобразовано к datatype типу данных.